src/solver.rs

```diff
@@ -0,0 +1,275 @@
+use crate::c_bindings;
+use crate::coords::{LLHRadians, ECEF};
+use crate::navmeas::NavigationMeasurement;
+use crate::signal::GnssSignal;
+use crate::time::GpsTime;
+use std::borrow::Cow;
+use std::ffi;
+use std::fmt;
+
+#[derive(Clone, Debug)]
+pub struct GnssSolution(c_bindings::gnss_solution);
+
+impl GnssSolution {
+    pub fn pos_valid(&self) -> bool {
+        self.0.valid == 1
+    }
+
+    pub fn vel_valid(&self) -> bool {
+        self.0.velocity_valid == 1
+    }
+
+    pub fn pos_llh(&self) -> LLHRadians {
+        LLHRadians::from_array(&self.0.pos_llh)
+    }
+
+    pub fn pos_ecef(&self) -> ECEF {
+        ECEF::from_array(&self.0.pos_ecef)
+    }
+
+    // pub fn vel_ned(&self) -> ECEF {
```

Comment:
?

Reply:
Good catch. I need to add a struct for NED coordinate data, at the moment we only have ECEF and LLH